prover/src/lib.rs

```diff
@@ -191,14 +199,28 @@ pub trait Prover {
     // PROVIDED METHODS
     // --------------------------------------------------------------------------------------------
 
+    /// Builds the auxiliary trace.
+    #[allow(unused_variables)]
+    fn build_aux_trace<E, Hasher>(
+        &self,
+        main_trace: &ColMatrix<E::BaseField>,
+        transcript: &mut impl RandomCoin<BaseField = E::BaseField, Hasher = Hasher>,
+    ) -> AuxTraceWithMetadata<E, AuxRandElementsProver<Self, E>, AuxProofProver<Self>>
```

Comment:
Can we pass `main_trace` into this as `Self::Trace`?

Also, is there a reason why we pass `RandomCoin` as `impl` rather than as `Self::RandomCoin`?

Lastly, I think we should update the comment here to describe in more detail what this method is supposed to do in various cases.

Reply:
This was an artifact of copy/pasting from the former `AuxTraceBuilder`. Now fixed.

I also documented what an auxiliary proof is in `AuxProofWithMetadata`, and `build_aux_trace()` now refers to it "for more information".